server/src/main/java/io/crate/execution/engine/indexing/IndexWriterProjector.java

```diff
@@ -188,9 +188,9 @@ public String value() {
             }
             assert value instanceof LinkedHashMap<String, Object> : "the raw source order should be preserved";
             Map<String, Object> filteredMap = XContentMapValues.filter(value, includes, excludes);
-            try {
-                BytesReference bytes = BytesReference.bytes(new XContentBuilder(XContentType.JSON.xContent(),
-                    new BytesStreamOutput(lastSourceSize)).map(filteredMap));
+            try (var stream = new BytesStreamOutput(lastSourceSize)) {
+                XContentBuilder xContentBuilder = new XContentBuilder(XContentType.JSON.xContent(), stream);
+                BytesReference bytes = BytesReference.bytes(xContentBuilder.map(filteredMap));
```

Comment:
This `XContentBuilder` usage brings to 
`public XContentBuilder(XContent xContent, OutputStream bos)` which [says](public XContentBuilder(XContent xContent, OutputStream bos) ) we must close builder.

I think we need to put xContentBuilder to the try-with-resources.

Reply:
checked other usages, looks like we close it everywhere, in the indexers `BytesReference.bytes(xContentBuilder)` closes the builder.

Reply:
It's the same here as well. `BytesReference.bytes(xContentBuilder)` closes it.